web-app/test/unit/resources/elements/panels/main-panel-spec.js

```diff
@@ -114,34 +114,80 @@ describe('MainPanel', () => {
 
     describe('setScalingFactor', () => {
 
+        const dataUri = 'data:image/jpeg;base64,R0lGODlhkAGQAfcAAAQCBJKDLJSSlExGHNS/NNTOtFRSVCckFGxjJPjgXMzKzHyCfKuqpDMxNPzu0fXaJGxqZLikLEdCPBQSDDQyFOvq7H90LL6PlFxOw==';
+
+        const mockImage = {
+            src: null,
+            onload: () => {}
+        };
+
+        const realImage = global.Image;
+
+        beforeAll(() => {
+            global.Image = () => { return mockImage; };
+        });
+
+        afterAll(() => {
+            global.Image = realImage;
+        })
+
         beforeEach(() => {
-            // need to mock the config setting: this.options.image.fitImageToWindow
-            mainpanel.offsetWidth = 100;
-            mainpanel.offsetHeight = 100;
+            mainpanel.scalingFactor = 2.0; // dummy factor
+            mainpanel.options = {};
+            mainpanel.dataURI = dataUri;
         });
 
-        it('verify minimum zoom of 0.125', async () => {
-            // mock image dimension
+        it('verify no scaling calculation without option set', () => {
```

Comment:
Simple test example but you will notice I call onload on our mock AFTER the call to setScalingFactor.  technically we could have modified the src set/get via a defineProperty( ) call to auto-call the onload when the src is set but we have to remember that image loading is asynchronous.  So when you set a Image.src you are telling the element to load the image and once loaded it calls onload.  Since we mocked image this never really happens, but we can trigger it and then observe the outcome of it.  This took me a while to get right which is another reason to split the calculation code into a submethod and test that cleanly for the conditions and then is more testing for the "affect" of did it get called on load.  I am thinking I may still want to try and auto-invoke the onload on a src property being set so you don't have to manually call it to load....